iocore/net/SSLNetVConnection.cc

```diff
@@ -2089,15 +2088,15 @@ SSLNetVConnection::_ssl_connect()
 
       Debug("ssl.origin_session_cache", "origin session cache lookup key = %s", lookup_key.c_str());
 
-      sess = this->getOriginSession(ssl, lookup_key);
+      std::shared_ptr<SSL_SESSION> shared_sess = this->getOriginSession(ssl, lookup_key);
+      sess                                     = shared_sess.get();
```

Comment:
Why do this? It defeats the point of a `shared_ptr`.

Reply:
done